# Coding Exercises (Part 1)

Now, you will have the opportunity to practice what you have learned. <br>
__Follow the instructions__ and insert your code!

The correct answer is provided below your coding cell. There you can check whether your code is correct.

If you need some help or if you want to check your code, you can also have a look at the __solutions__.

### Have Fun!

--------------------------------------------------------------------------------------------------------------

## Exercise 1: Using Python as a Calculator for TVM Problems

Use __variables__ where appropriate! __Input__ values and your __final results__ shall be stored in variables.

1. You save 2,000 USD for 7 years at an interst rate of 4% p.a.. Calculate the value of your account in 7 Years (__Future Value__)!

In [ ]:
# Store Input values in variables
pv = 2000 # present value
n = 7 # number of years
r = 0.04 # interest rate


In [ ]:
# Calculate and store the future value in a variable
fv = pv * (1+r)**n # future value

In [ ]:
# print the future value
print(round(fv,2))

2631.86


(result: 2631.86 USD)

2. You consider to purchase a car in 4 years for 20,000 USD. How many USD do you have to save today (__Present Value__) in your savings account to have 20,000 USD in 4 years? Assume an interst rate of 3.5% p.a.

In [ ]:
# Store Input values in variables
fv = 20000 # car price
n = 4 # number of years
r = 0.035 # interest rate

In [ ]:
# Calculate and store the present value in a variable
pv = fv / ((1+r)**n)

In [ ]:
# print the present value
print(round(pv,2))

17428.84


(result: 17,428.84 USD)

3. Bank B offers you to save 10,000 USD today, getting back 15,000 USD in 10 Years. Calculate the __per annum compound interest rate__. Bank A offers you an interest rate of 4.5% p.a.. Which offer should you accept?

In [ ]:
pv = 10000
fv = 15000
n  = 10
r_A = 0.045 # interest rate of the bank A

In [ ]:
r_B = (fv/pv)**(1/n) - 1

In [ ]:
print(round(r_B*100,2))

4.14


(result: 4.14%)

* Which offer should you accept?: Bank A offers the better interest rate.

4. One year ago you invested 976 USD in the ABC stock. Recently you received a 40 USD Dividend Payment and the current value of your ABC stock investment is 1,053 USD. Calculate the __Price Return__!

In [ ]:
p0 = 976
dividend = 40
p1 = 1053
stock_ret = (p1/p0) - 1
stock_ret = round(stock_ret*100,2)

In [ ]:
print(stock_ret)

7.89


(result: 7.89%)

5. Calculate __Dividend Yield__ and __Total Return__ for your ABC stock investment!

In [ ]:
dividend_yield = dividend / p0
dividend_yield = round(dividend_yield*100 ,2)

In [ ]:
print(dividend_yield)

4.1


In [ ]:
total_ret = stock_ret + dividend_yield
total_ret =  round(total_ret,2)

In [ ]:
print(total_ret)

11.99


(results: 4.10% and 11.99%)

6. Print Price Return, Dividend Yield and Total Return in one single line, __separated by commas__!

In [ ]:
print(f"Price Return: {ret}, Dividend Yield : {dividend_yield} , Total Return: {total_ret}")

Price Return: 7.89, Dividend Yield : 4.1 , Total Return: 11.99


# Well Done!

---------------------------------------------------------------------------------------------------------------------